### PR TITLE
Makes the pda font a prefernce

### PR DESCRIPTION
**Pull request by Supermichael777**
Its now a preference applied only on a pda's first equip.
:cl:
tweak: The PDA default font has been switched from "eye bleed" to old-style monospace. Check your preference.
/:cl:

Muh eyes.

pda

VT323 (old new default)
image

Share Tech Mono
image

Orbitron
image

Monospaced (new new default)
image

### CODE REVIEW

**Comment by CPTANT** [comment]
pics

**Comment by Cyberboss** [comment]
Instead of bikeshedding over which font is better, make it a preference

**Comment by ShizCalev** [comment]
bless

**Comment by Supermichael777** [comment]
Makeing this a preference is going to be hard. Let me see if the equip will support it.

and this is the one styled like the old pda font.

**Comment by Supermichael777** [comment]
QUESTION TIME: Should the font be set only the first time it is equipped or any time its picked up?
and should the choice be persistent(fuck sql)?

**Comment by Bawhoppen** [comment]
It should be an in-game PDA setting

**Comment by Supermichael777** [comment]
it is and will stay but from the context I think your going to want it set on first equip (which would be startup or when its pulled from the box-o pdas or lootcrate). That's the direction I was heading.

**Comment by Bawhoppen** [comment]
You could make it randomized so everyone's equally unhappy

**Comment by Supermichael777** [comment]
randomized is even harder, you all get mono

**Comment by nicbn** [comment]
Just remove PDAs and everyone is happy.

**Comment by Bawhoppen** [comment]
Now someone's talking the right idea

**Comment by SgtxRootbeer** [comment]
Make it and the font, and the ringtone preferences.

**Comment by praisenarsie** [comment]
can you spelllllllllll review

**Comment by ChangelingRain** [comment]
this should be a var on the PDA itself

**Comment by ChangelingRain** [comment]
like, your preference should only decide what font your PDA starts with, there should be an option in the PDA to change it on top of that

**Comment by KorPhaeron** [comment]
Preference bloat is terrible

**Comment by optimumtact** [comment]
Do you even know how to code lol

**Comment by Supermichael777** [comment]
It is a var on the pda, this only applies the first time, Preference bloat is nt natural result of reducing bikesheading about what should be the default user experence and how it should be set, We have like 10 preferences that actually matter

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -54,6 +54,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 	var/detonatable = TRUE // Can the PDA be blown up?
 	var/hidden = 0 // Is the PDA hidden from the PDA list?
 	var/emped = 0
+	var/equiped = 0 //is this the first time it has been equiped?
```
Comment (lzimann) [inline code/game/objects/items/devices/PDA/PDA.dm]: `FALSE`

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -87,6 +88,27 @@ GLOBAL_LIST_EMPTY(PDAs)
...
+				if(ORBITRON)
+					font_index = 2
+					font_mode = "font-family:\"Orbitron\", monospace;letter-spacing:0px; font-size:15px"
+				if(VT)
+					font_index = 3
+					font_mode = "font-family:\"VT323\", monospace;letter-spacing:1px;"
+				else
+					font_index = 0
+					font_mode = "font-family:monospace;"
+		equiped = 1
```
Comment (lzimann) [inline code/game/objects/items/devices/PDA/PDA.dm]: `TRUE`

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 3]
```diff
@@ -87,6 +88,27 @@ GLOBAL_LIST_EMPTY(PDAs)
 		inserted_item =	new /obj/item/pen(src)
 	update_icon()
 
+/obj/item/device/pda/equipped(mob/user, slot)
+	if(!equiped)
+		if(user.client)
+			switch(user.client.prefs.pda_style)
+				if(MONO)
+					font_index = 0
```
Comment (lzimann) [inline code/game/objects/items/devices/PDA/PDA.dm]: use the font defines

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 4]
```diff
@@ -87,6 +88,27 @@ GLOBAL_LIST_EMPTY(PDAs)
...
+/obj/item/device/pda/equipped(mob/user, slot)
+	if(!equiped)
+		if(user.client)
+			switch(user.client.prefs.pda_style)
+				if(MONO)
+					font_index = 0
+					font_mode = "font-family:monospace;"
+				if(SHARE)
+					font_index = 1
+					font_mode = "font-family:\"Share Tech Mono\", monospace;letter-spacing:0px;"
```
Comment (lzimann) [inline code/game/objects/items/devices/PDA/PDA.dm]: the names could also use the proper defines

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (kevinz000) [inline code/game/objects/items/devices/PDA/PDA.dm]: spelling

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 4 above)
Comment (Supermichael777) [reply code/game/objects/items/devices/PDA/PDA.dm]: the names are defined but for the player facing side, this is dynamic html code

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (Supermichael777) [reply code/game/objects/items/devices/PDA/PDA.dm]: finneee

**Pull request merged**